snapcast/control/server.py

```diff
@@ -442,11 +446,18 @@ def _on_stream_update(self, data):
             if data.get('stream', {}).get('uri', {}).get('query', {}).get('codec') == 'null':
                 _LOGGER.debug('stream %s is input-only, ignore', data.get('id'))
             else:
-                _LOGGER.info('stream %s not found, synchronize', data.get('id'))
-
-                async def async_sync():
-                    self.synchronize((await self.status())[0])
-                asyncio.ensure_future(async_sync())
+                self._synchronize_if_stream_missing(data.get('id'))
+
+    def _synchronize_if_stream_missing(self, stream_id):
+        """Ensure stream exists, otherwise synchronize."""
+        if stream_id is None:
+            return
+        if stream_id not in self._streams:
+            _LOGGER.info('stream %s not found, synchronize', stream_id)
+
+            async def async_sync():
+                self.synchronize((await self.status())[0])
```

Comment:
Call on_update callback:
```suggestion
                self.synchronize((await self.status())[0])
                if self._on_update_callback_func and callable(self._on_update_callback_func):
                    self._on_update_callback_func()
```